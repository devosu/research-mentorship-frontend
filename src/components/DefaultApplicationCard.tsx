// @components/DefaultMentorCard
//
// DefaultMentorCard component definition.

// NextJS essential imports.

// Local imports.
 import { db } from '@src/lib/firebaseInit';
 import { doc, setDoc, getDoc, collection, serverTimestamp, updateDoc } from "firebase/firestore"; 

interface Props {
    id: string,
    name?: string
    major?: string
    desiredResearchField?: string
    type?: string
    currentResearchField?: string
    currentResearchFieldDescription?: string
  }

export default function DefaultApplicationCard (props: Props) {

    async function acceptApplication() {
        const docRef = doc(db, "applications", props.id);
        await updateDoc(docRef, {
            adminApproved: true
          });
    }

    async function rejectApplication() {
        const docRef = doc(db, "applications", props.id);
        await updateDoc(docRef, {
            adminApproved: false
          });
    }

    return (
            <div className="bg-white shadow-md rounded-lg p-6">
                <h2 className="text-xl font-semibold">Name: {props.name}</h2>
                <p className="mt-2">type: {props.type}</p>
                <p className="mt-2">Major: {props.major}</p>
                <p className="mt-2">Desired Research Field: {props.currentResearchField}</p>
                <button
                onClick={acceptApplication}
                type="button"
                className="inline-block rounded bg-primary px-6 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-white shadow-primary-3 transition duration-150 ease-in-out hover:bg-primary-accent-300 hover:shadow-primary-2 focus:bg-primary-accent-300 focus:shadow-primary-2 focus:outline-none focus:ring-0 active:bg-primary-600 active:shadow-primary-2 motion-reduce:transition-none dark:shadow-black/30 dark:hover:shadow-dark-strong dark:focus:shadow-dark-strong dark:active:shadow-dark-strong">
                Accept
                </button>
                <button
                onClick={rejectApplication}
                type="button"
                className="inline-block rounded bg-primary px-6 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-white shadow-primary-3 transition duration-150 ease-in-out hover:bg-primary-accent-300 hover:shadow-primary-2 focus:bg-primary-accent-300 focus:shadow-primary-2 focus:outline-none focus:ring-0 active:bg-primary-600 active:shadow-primary-2 motion-reduce:transition-none dark:shadow-black/30 dark:hover:shadow-dark-strong dark:focus:shadow-dark-strong dark:active:shadow-dark-strong">
                Reject
                </button>
            </div>
    );
    }