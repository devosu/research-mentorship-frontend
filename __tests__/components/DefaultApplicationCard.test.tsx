// ./__tests__/components/DefaultApplicationCard.test.tsx
//
// Snapshot tests for the DefaultApplicationCard component.

// jsdom-jest essential imports.
import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";

// Local testing module imports.
import DefaultApplicationCard, {
  type Props,
} from "@components/DefaultApplicationCard";

// Behavior for the DefaultApplicationCard component.
describe("The default application card component", () => {
  // Setup a test prop to render the DefaultApplicationCard component.
  const testProps: Props = {
    id: "1",
    name: "John Doe",
    major: "Computer Science",
    desiredResearchField: "Machine Learning",
    type: "Undergraduate",
    currentResearchField: "Computer Vision",
    currentResearchFieldDescription:
      "I am currently researching computer vision.",
  };

  it("renders a div element", () => {
    const { container } = render(
      <DefaultApplicationCard
        id={testProps.id}
        name={testProps.name}
        major={testProps.major}
        desiredResearchField={testProps.desiredResearchField}
        type={testProps.type}
        currentResearchField={testProps.currentResearchField}
        currentResearchFieldDescription={
          testProps.currentResearchFieldDescription
        }
      />,
    );
    expect(container.querySelector("div")).toBeDefined();
  });

  it("matches the snapshot", () => {
    const { container } = render(
      <DefaultApplicationCard
        id={testProps.id}
        name={testProps.name}
        major={testProps.major}
        desiredResearchField={testProps.desiredResearchField}
        type={testProps.type}
        currentResearchField={testProps.currentResearchField}
        currentResearchFieldDescription={
          testProps.currentResearchFieldDescription
        }
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
