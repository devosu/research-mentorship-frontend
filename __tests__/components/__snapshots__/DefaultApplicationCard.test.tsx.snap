// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The default application card component matches the snapshot 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded-lg p-6"
  >
    <h2
      class="text-xl font-semibold"
    >
      Name: 
      John Doe
    </h2>
    <p
      class="mt-2"
    >
      type: 
      Undergraduate
    </p>
    <p
      class="mt-2"
    >
      Major: 
      Computer Science
    </p>
    <p
      class="mt-2"
    >
      Desired Research Field: 
      Computer Vision
    </p>
    <button
      class="inline-block rounded bg-primary px-6 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-white shadow-primary-3 transition duration-150 ease-in-out hover:bg-primary-accent-300 hover:shadow-primary-2 focus:bg-primary-accent-300 focus:shadow-primary-2 focus:outline-none focus:ring-0 active:bg-primary-600 active:shadow-primary-2 motion-reduce:transition-none dark:shadow-black/30 dark:hover:shadow-dark-strong dark:focus:shadow-dark-strong dark:active:shadow-dark-strong"
      type="button"
    >
      Accept
    </button>
    <button
      class="inline-block rounded bg-primary px-6 pb-2 pt-2.5 text-xs font-medium uppercase leading-normal text-white shadow-primary-3 transition duration-150 ease-in-out hover:bg-primary-accent-300 hover:shadow-primary-2 focus:bg-primary-accent-300 focus:shadow-primary-2 focus:outline-none focus:ring-0 active:bg-primary-600 active:shadow-primary-2 motion-reduce:transition-none dark:shadow-black/30 dark:hover:shadow-dark-strong dark:focus:shadow-dark-strong dark:active:shadow-dark-strong"
      type="button"
    >
      Reject
    </button>
  </div>
</div>
`;
